### PR TITLE
[gh-249] Fix infinity loop on Upgrade stack

### PR DESCRIPTION
**Pull request by andyduong1920**
https://github.com/nimblehq/elixir-templates/issues/249

## What happened 👀

Solved https://github.com/nimblehq/elixir-templates/issues/249

## Insight 📝

It's because the `anchor is equal to the content`, so that method keep loop forever.

In our case, it's because we have the `-otp-25` in existing Elixir version, and when we upgrade to another `erlang 25` version, we try to `find and replace the -otp-25 by -otp-25`

image

## Proof Of Work 📹

1. Tested on my local
2. Trigger the Workflow https://github.com/nimblehq/elixir-templates/actions/runs/3426992297/jobs/5709447990
3. We have the PR created by that workflow https://github.com/nimblehq/elixir-templates/pull/286

### CODE REVIEW

File: lib/nimble_template/helpers/generator.ex [hunk 1]
```diff
@@ -24,7 +24,12 @@ defmodule NimbleTemplate.Generator do
 
   def rename_file(old_path, new_path), do: File.rename(old_path, new_path)
 
-  def replace_content(file_path, anchor, content, raise_exception \\ true) do
+  def replace_content(file_path, anchor, content, raise_exception \\ true)
+
+  def replace_content(_file_path, anchor, content, _raise_exception) when anchor == content,
```
Comment (liamstevens111) [inline lib/nimble_template/helpers/generator.ex:29]:
```suggestion
  def replace_content(_file_path, anchor, anchor, _raise_exception),
```
I think this also works but yours I think is more clear what it's doing? 👁️

**Review by liamstevens111 — APPROVED** [review]
Comment/Question but nice work 👍

File: lib/nimble_template/helpers/generator.ex (hunk 1 above)
Comment (andyduong1920) [reply lib/nimble_template/helpers/generator.ex:29]: that isn't clear enough I think @liamstevens111 and I'm not sure it work as the same 😂 so let go with `when anchor == content`

Comment (liamstevens111) [reply lib/nimble_template/helpers/generator.ex:29]: Yeah should work the same but I agree, probably better to be explicit here @andyduong1920

**Review by byhbt — APPROVED** [review]

**Pull request merged**